src/VisualStudio/TestUtilities/OutOfProcess/Editor_OutOfProc.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Roslyn.VisualStudio.Test.Utilities.Common;
+using Roslyn.VisualStudio.Test.Utilities.InProcess;
+using Roslyn.VisualStudio.Test.Utilities.Input;
+
+namespace Roslyn.VisualStudio.Test.Utilities.OutOfProcess
+{
+    /// <summary>
+    /// Provides a means of interacting with the Visual Studio editor by remoting calls into Visual Studio.
+    /// </summary>
+    public partial class Editor_OutOfProc : OutOfProcComponent<Editor_InProc>
+    {
+        internal Editor_OutOfProc(VisualStudioInstance visualStudioInstance)
+            : base(visualStudioInstance)
+        {
+        }
+
+        public void Activate() => InProc.Activate();
+
+        public string GetText() => InProc.GetText();
+        public void SetText(string value) => InProc.SetText(value);
+
+        public string GetCurrentLineText() => InProc.GetCurrentLineText();
+        public int GetCaretPosition() => InProc.GetCaretPosition();
+        public string GetLineTextBeforeCaret() => InProc.GetLineTextBeforeCaret();
+        public string GetLineTextAfterCaret() => InProc.GetLineTextAfterCaret();
+
+        public void MoveCaret(int position) => InProc.MoveCaret(position);
+
+        public string[] GetCompletionItems() => InProc.GetCompletionItems();
+        public string GetCurrentCompletionItem() => InProc.GetCurrentCompletionItem();
+        public bool IsCompletionActive() => InProc.IsCompletionActive();
+
+        public Signature[] GetSignatures() => InProc.GetSignatures();
+        public Signature GetCurrentSignature() => InProc.GetCurrentSignature();
+
+        public bool IsCaretOnScreen() => InProc.IsCaretOnScreen();
+
+        public void SendKeys(params object[] keys)
```

Comment:
Doc the allowed types for keys here? I love how this does the virtual key stuff; it's just a bit obtuse for what you can give it.